packages/react-vapor/src/components/table-hoc/TableSelectors.ts

```diff
@@ -19,6 +20,17 @@ const initialTableSort: ITableWithSortState = {
 const getIsEmpty = (state: IReactVaporState, props: TableSelectorsProps): boolean =>
     props.data !== null && (!props.data || props.data.length === 0);
 
+const getIsTruelyEmpty = (state: IReactVaporState, props: TableSelectorsProps): boolean => {
+    const compositeState = TableHOCUtils.getCompositeState(props.id, state);
+    const isEmpty = getIsEmpty(state, props);
+
+    const hasNoAppliedFilter = !compositeState.filter;
+    const hasNoAppliedPredicates = !(compositeState.predicates?.length > 0);

```

Comment:
[optional] maybe `const hasNoAppliedPredicates = !compositeState.predicates?.length;` since 0 is falsy